src/DynamoApplications/StartupUtils.cs

```diff
@@ -339,6 +349,25 @@ public static DynamoModel MakeCLIModel(string asmPath, string userDataFolder, st
             return model;
         }
 
+        /// <summary>
+        /// Use this overload to construct a DynamoModel in CLI context when the location of ASM to use is known, host analytics info is known and you want to set data paths.
+        /// </summary>
+        /// <param name="asmPath">Path to directory containing geometry library binaries</param>
+        /// <param name="userDataFolder">Path to be used by PathResolver for UserDataFolder</param>
+        /// <param name="commonDataFolder">Path to be used by PathResolver for CommonDataFolder</param>
+        /// <param name="info">Host analytics info specifying Dynamo launching host related information.</param>
+        /// <param name="isServiceMode">Boolean indication of launching Dynamo in service mode, this mode is optimized for minimal launch time.</param>
+        /// <returns></returns>
+        public static DynamoModel MakeCLIModel(string asmPath, string userDataFolder, string commonDataFolder, HostAnalyticsInfo info = new HostAnalyticsInfo(), bool isServiceMode = false)
```

Comment:
I have also added a couple of flags last year :). I guess we need to think of something soon to better consolidate all the MakeModel constructs. Perhaps a startup config file where you can enable/disable features/extensions etc.

Reply:
agreed, currently the code (constructors) used for CLI are a bit messy